File: 150-Evaluate-Reverse-Polish-Notation.py
class Solution(object):
    def calc(slef, op, x, y):
        if op == "+":
            return x + y
        elif op == "-":
            return x - y
        elif op == "*":
            return x * y
        elif op == "/":
            return int(float(x) / y)

    def evalRPN(self, tokens):
        stack = []

        for token in tokens:
            if token.isdigit() or (token[0] == "-" and token[1:].isdigit()):
                stack.append(int(token))
            else:
                y = stack.pop()
                x = stack.pop()
                stack.append(self.calc(token, x, y))

        return stack[0]

File: neetcode/150-Evaluate-Reverse-Polish-Notation.ts
function isNum(c: string): boolean {
  return !isNaN(+c);
}

function evalRPN(tokens: string[]): number {
  const stack: number[] = [];

  for (const token of tokens) {
    if (isNum(token)) {
      stack.push(+token);
    } else {
      const top1 = stack.pop();
      const top2 = stack.pop();

      if (token === '*') {
        stack.push(top2 * top1);
      } else if (token === '+') {
        stack.push(top2 + top1);
      } else if (token === '-') {
        stack.push(top2 - top1);
      } else if (token === '/') {
        stack.push(Math.trunc(top2 / top1));
      }
    }
  }

  return stack[0];
}